tests/models/common_chat_testing.py

```diff
@@ -14,8 +15,15 @@ def smoke_chat(lm: models.Chat, has_system_role: bool = True):
         lm += "Pick a number: "
 
     print(str(lm))
+    print(f"{lm.engine.metrics=}")
     assert len(lm["text"]) > 0
     assert str(lm).endswith("Pick a number: <|im_end|>")
+    assert lm.engine.metrics.engine_input_tokens > 2, "Expect some input tokens"
```

Comment:
These are _very_ simple tests of the functionality. They at least show that tokens are being counted.